### PR TITLE
Provide a semantic error in Gatsby-Image components to non-TypeScript users when `onClick` props are added

### PR DESCRIPTION
**Pull request by hu0p**
## Description

This throws an error during the render method of gatsby-image components if someone tries to add an invalid `onClick` prop. `onClick` props were removed in https://github.com/gatsbyjs/gatsby/pull/19458 for accessibility reasons. The version was only incremented for a bug fix, but it should've been at least a minor version change due to it being breaking. 

I would've incremented the version myself, but I think Lerna handles in some fashion so I left it alone.

## Related Issues

https://github.com/gatsbyjs/gatsby/pull/19458 and https://github.com/gatsbyjs/gatsby/issues/18468

### CODE REVIEW

File: packages/gatsby-image/src/index.js [hunk 1]
```diff
@@ -414,6 +414,11 @@ class Image extends React.Component {
       itemProp,
     }
 
+    if (this.props.onClick)
+      throw new Error(
+        `'onClick' event handlers are not allowed for accessibility reasons.`
+      )
+
```
Comment (pvdz) [inline packages/gatsby-image/src/index.js]: Code style: Please wrap sub-statements in a block if they do not stay on the same line. Prettier won't auto-fix this for you (although we should add a linting rule for this, if one isn't already in place).

Doing it without a block can be confusing in the JS world and in particular to cases where a newline is forbidden (`return` arg, `throw`, `yield`).

Comment (hu0p) [reply packages/gatsby-image/src/index.js]: Understood. I've gone ahead and wrapped it in a block. Is it alright to `throw` without `return`, or would it be best to change that too?

Comment (pvdz) [reply packages/gatsby-image/src/index.js]: A `throw` is effectively a hard `return` so that's not necessary :)

**Comment by pvdz** [comment]
Thank you for your suggestion in this PR. And for improving the style issue I brought up.

However, while we do agree that this would conceptually be a good idea, we are considering a more generic approach to improve the a11y situation around this, rather than a one-off fix. As such I'm going to close this for now.

Thanks again!

**Comment by hu0p** [comment]
@pvdz Sounds great! What is that generic approach? As I [noted](https://github.com/gatsbyjs/gatsby/pull/19458#issuecomment-557180513 ), this isn't really my go-to solution to the problem. I'd much rather make it possible for gatsby-image to be both clickable and accessible, but this was what I was able to contribute with limited time. I'd love to contribute to a better approach or participate in the conversation surrounding it. Could you point me to where that's happening?

**Comment by JaKXz** [comment]
> we are considering a more generic approach to improve the a11y situation around this

@pvdz is there an issue to track this work?

**Comment by pvdz** [comment]
Best reference I have right was already mentioned in the PR; #18468

**Comment by JaKXz** [comment]
That issue mentions rolling back this functionality but does not talk about the generic approach you mentioned here.

**Comment by hu0p** [comment]
@pvdz Can we talk about the breaking change this introduced? I feel this merited a major version bump. [Semantic versioning](https://semver.org/#spec-item-8) states that the major version must be incremented when backwards incompatible changes are introduced. Various popular dev blogs (some of which are even built with Gatsby) reiterate this as an accepted standard. Does Gatsby follow a different standard?

**Comment by JaKXz** [comment]
cc @KyleAMathews @wardpeet sorry to tag you directly but I think this is an important question ^^

**Pull request closed without merging**